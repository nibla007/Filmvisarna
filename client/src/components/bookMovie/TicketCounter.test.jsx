import { render, fireEvent, screen } from "@testing-library/react";
import TicketCounter from "./TicketCounter";
import renderer from "react-test-renderer";

jest.mock('../../react-easier', () => ({
  useStates: jest.fn(() => ({ adult: 0, child: 1, senior: 2})), // Sets value of the amount of people
}));

  describe('TicketCounter', () => {
    test('renders with initial values', () => {
      const setSeatsMock = jest.fn();
      render(
        <TicketCounter
          screening={{}}
          movie={{}}
          seats={[]}
          setSeats={setSeatsMock}
        />
      );

      // Check if mockdata values are in the document
      expect(screen.getByText('0')).toBeInTheDocument(); // adult
      expect(screen.getByText('1')).toBeInTheDocument(); // child
      expect(screen.getByText('2')).toBeInTheDocument(); // senior
    });

    test("snapshot testing", () => {
      const setSeatsMock = jest.fn();
      const snap = renderer.create(<TicketCounter
          screening={{}}
          movie={{}}
          seats={[]}
          setSeats={setSeatsMock}
        />).toJSON();
      expect(snap).toMatchSnapshot("");
    });
  });

  


