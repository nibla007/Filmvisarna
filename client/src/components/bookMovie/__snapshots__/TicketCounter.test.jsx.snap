// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  className="flex flex-col justify-evenly p-8 text-center text-xl text-white sm:mt-40 sm:flex-row sm:text-2xl lg:w-[80%]"
>
  <div
    className="sm:text-3xl lg:w-2/5"
  >
    <h2
      className="mb-6 text-3xl sm:text-3xl lg:text-left lg:text-[2.5rem]"
    >
      Välj antal biljetter
    </h2>
    <div
      className="lg:flex lg:items-center lg:justify-between"
    >
      <p
        className="sm:text-2xl"
      >
        Vuxen (140 kr)
      </p>
      <div
        className=" flex items-center justify-center"
      >
        <button
          className="text-4xl sm:text-5xl"
          name="adult"
          onClick={[Function]}
        >
          -
        </button>
        <p
          className="mx-3 text-2xl sm:text-3xl"
        >
          0
        </p>
        <button
          className="text-3xl sm:text-4xl"
          name="adult"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
    <div
      className="lg:flex lg:items-center lg:justify-between"
    >
      <p
        className="sm:text-2xl"
      >
        Barn (120 kr)
      </p>
      <div
        className="flex items-center justify-center"
      >
        <button
          className="text-4xl sm:text-5xl"
          name="child"
          onClick={[Function]}
        >
          -
        </button>
        <p
          className="mx-3 text-2xl sm:text-3xl"
        >
          1
        </p>
        <button
          className="text-3xl sm:text-4xl"
          name="child"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
    <div
      className="lg:flex lg:items-center lg:justify-between"
    >
      <p
        className="sm:text-2xl"
      >
        Pensionär (80 kr)
      </p>
      <div
        className="flex items-center justify-center"
      >
        <button
          className="text-4xl sm:text-5xl"
          name="senior"
          onClick={[Function]}
        >
          -
        </button>
        <p
          className="mx-3 text-2xl sm:text-3xl"
        >
          2
        </p>
        <button
          className="text-3xl sm:text-4xl"
          name="senior"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
  </div>
  <div
    className="mt-6 flex gap-6"
  >
    <img
      alt="movie poster"
      className="h-44 sm:mt-auto sm:h-60"
      src="/img/undefined"
    />
    <div
      className="flex flex-col justify-end text-start text-base"
    >
      <p
        className="text-xl"
      />
      <p>
        Invalid Date:e undefined | undefined
      </p>
      <p />
      <p />
      <p>
         
      </p>
      <p>
         undefined år
      </p>
    </div>
  </div>
</div>
`;
